### ISSUE TITLE
How to detection length of an object?

### ISSUE THREAD

**Issue by biswajitschemaphic** [ultralytics/ultralytics#18258]
### Search before asking

- [X] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

Hello @glenn-jocher , you are very knowledgeable, informative and helpful!

Will you kindly help me how can I detect length of an iron billet? I have made a custom yolov8 model of the iron billets. But I want to detect the length of each detected billet as well. Example image attached.

Will you kindly give me an example script where it can successfully detect the length of an object from cctv footage? Also is it possible to detect weight of the object?

![Screenshot 2024-12-16 174435](https://github.com/user-attachments/assets/627903e0-d16e-4db3-a4b1-f32c7b9339b0)

**Comment by UltralyticsAssistant** [comment]
👋 Hello @biswajitschemaphic, thank you for your interest in Ultralytics 🚀! We recommend starting with the [Ultralytics Docs](https://docs.ultralytics.com), where you can find various guides and examples for using YOLO models with Python and CLI, as well as tips for custom object detection tasks.

To address your question on measuring the length of an object, while YOLO models primarily detect objects and return bounding boxes, calculating physical dimensions such as length depends on additional factors like camera calibration and pixel-to-real-world scaling. We encourage you to review our [Custom Training Tips](https://docs.ultralytics.com/guides/model-training-tips/) to ensure the best performance for your dataset and consider researching methods for determining lengths or dimensions from detected object bounding boxes.

If this is a 🐛 Bug or an ❓ implementation question, please clarify further. For bugs, providing a [minimum reproducible example](https://docs.ultralytics.com/help/minimum_reproducible_example/) can greatly assist our debugging process.

### Community Support

We also suggest sharing your question and learning from the broader Ultralytics community:
- Join our [Discord community](https://discord.com/invite/ultralytics) for real-time discussions 🎧
- Start a topic on the [Discourse forum](https://community.ultralytics.com) for detailed troubleshooting
- Post on our [Subreddit](https://reddit.com/r/Ultralytics) to share experiences and solutions with others

### Upgrade

Ensure that you have the latest tools to avoid encountering resolved issues. You can update the `ultralytics` package and verify you meet the required dependencies by running the following command in a [**Python>=3.8**](https://www.python.org/) environment:

```bash
pip install -U ultralytics
```

### Environments

Here are some pre-configured environments where you can test, train, and deploy YOLO models:

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Cloud options**, such as **Google Cloud** ([GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)) and **Amazon** ([AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/))
- **Docker**: Efficiently manage and deploy your projects. See our [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule) tests are currently passing. CI tests ensure successful execution of YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) across different platforms.

This is an automated response 🌟, but don't worry—an Ultralytics team member will review your issue and assist you soon.

**Comment by Skillnoob** [comment]
If you look directly straight at the object, then you can use the width of the detected box to calculate a estimated width of the object.

**Comment by biswajitschemaphic** [comment]
> If you look directly straight at the object, then you can use the width of the detected box to calculate a estimated width of the object.

Hello @Skillnoob, your reply is not clear for me. The attached image captured from cctv camera as an example. My requirement is to detect the exact length of this iron billets directly from the rtsp live link where the iron billets moving through out the frames continuously one by one. I hope it will make sense for you now.

**Comment by Skillnoob** [comment]
Yes, thats why I said that. From that perspective you'd need to do quite a bit of math to get the length of the object from just a box.

**Comment by biswajitschemaphic** [comment]
> Yes, thats why I said that. From that perspective you'd need to do quite a bit of math to get the length of the object from just a box.

Okay, thanks for your replies @Skillnoob. Can you give me an example? Suppose in real world scenario the iron billet size is 6 meter. Then how will I calculate this?
